code/game/machinery/computer/communications.dm

```diff
@@ -698,6 +698,10 @@
 	var/input = stripped_input(user, "Please choose a message to announce to the station crew.", "What?")
 	if(!input || !user.canUseTopic(src, !issilicon(usr)))
 		return
+	if(!(user.can_speak())) //No more cheating, mime/random mute guy!
+		input = "..."

```

Comment:
Should inform the user that they are mute in this case.

Reply:
Done